syntax = "proto3";

package protodb.server;

import "server/database_create.proto";
import "server/database_delete.proto";
import "server/database_list.proto";
import "server/database_define.proto";

import "server/collection_create.proto";
import "server/collection_delete.proto";
import "server/collection_list.proto";
import "server/collection_define.proto";

import "server/find.proto";
import "server/insert.proto";

service ProtoDB {
  // Database commands
  rpc CreateDatabase (protodb.server.CreateDatabase) returns (protodb.server.CreateDatabaseResponse);
  rpc DeleteDatabase (protodb.server.DeleteDatabase) returns (protodb.server.DeleteDatabaseResponse);
  rpc ListDatabases (protodb.server.ListDatabases) returns (protodb.server.ListDatabasesResponse);
  rpc DefineDatabase (protodb.server.DefineDatabase) returns (protodb.server.DefineDatabaseResponse);

  // Collection commands
  rpc CreateCollection (protodb.server.CreateCollection) returns (protodb.server.CreateCollectionResponse);
  rpc DeleteCollection (protodb.server.DeleteCollection) returns (protodb.server.DeleteCollectionResponse);
  rpc ListCollections (protodb.server.ListCollections) returns (protodb.server.ListCollectionsResponse);
  rpc DefineCollection (protodb.server.DefineCollection) returns (protodb.server.DefineCollectionResponse);

  rpc Find (protodb.server.Find) returns (protodb.server.FindResponse);
  rpc Insert (protodb.server.Insert) returns (protodb.server.InsertResponse);
}
